.github/scripts/integration-test-matrix.js

```diff
@@ -0,0 +1,90 @@
+module.exports = ({ context }) => {
+  const defaultPythonVersion = "3.8";
+  const supportedPythonVersions = ["3.6", "3.7", "3.8", "3.9"];
+  const supportedAdapters = ["snowflake", "postgres", "bigquery", "redshift"];
+
+  if (context.eventName.includes("pull_request")) {
+    // if PR, generate matrix based on files changed and PR labels
+    const changes = JSON.parse(process.env.CHANGES);
+    const labels = context.payload.pull_request.labels.map(({ name }) => name);
+    console.log("labels", labels);
+    console.log("changes", labels);
+    const testAllLabel = labels.includes("test all");
+    const include = [];
+
+    for (const adapter of supportedAdapters) {
+      if (
+        changes.includes(adapter) ||
```

Comment:
What format is `changes` in? Is this looking for the adapter name in a list of files, commit messages, etc.?

Reply:
it is a list of warehouse adapter names, I added a comment above to clarify! I also added some comments in the workflow as well https://github.com/dbt-labs/dbt/blob/ff9c8c8d854ff59be17414b68f472df7d1ed58b2/.github/workflows/integration.yml#L79-L103